### PR TITLE
Remove MAX_CONFIGS limit using WM_MENUCOMMAND for menu messages

### PR DESCRIPTION
**Pull request by selvanair**
We currently use WM_COMMAND message which is delivered with the
ID of the menu item requiring a unique ID for every command
(connect, disconnect etc..) for each connection profile. Instead,
use WM_MENUCOMMAND so that the message delivers a handle to the
menu and the position index of the menu item.

Connection menu array is now dynamically allocated. Yet, there
is still a limitation on the number of configs as the config
index + mgmt_port_offset must be < 65536 to be usable as a port
number. The error message shown for "too many configs" is reworded.
(English language file only).

Note: The current way of selecting the management port based on the
index of the config file increases chances of port conflicts
when the number of configs is large. It could be useful to change
this logic but that is beyond the cope of this PR.

Signed-off-by: Selva Nair <selva.nair@gmail.com>

### CODE REVIEW

File: tray.c [hunk 1]
```diff
@@ -71,6 +72,27 @@ CreateBitmaps()
     }
 }
 
+/*
+ * Reallocate hMenuConn as required. In case of not enough memory,
+ * o.num_config is reset to max space available value so that the
+ * program can continue.
+ */
+void
+allocate_connection_menu()
```
Comment (lstipakov) [inline tray.c:81]: All other functions in this file use CamelCase.

**Review by lstipakov — APPROVED** [review]
Looks good, tested with MSVC. One small nitpick.

File: tray.c (hunk 1 above)
Comment (selvanair) [reply tray.c:81]: Thanks for the review. Will change the function name for consistency.

**Comment by selvanair** [comment]
Renamed allocate_connection_menu() to AllocateConnectionMenu()

**Pull request merged**